File: recaudacion/2.Codigo/frontend/src/app/core/models/catalogobien.ts
import { UnidadMedida } from "./unidadmedida";

export class CatalogoBien {
    catalogoBienId: number;
    clasificadorIngresoId: number;
    unidadMedidaId: number;
    unidadMedida: UnidadMedida;
    codigo: string;
    descripcion: string;
    stockMaximo: number;
    stockMinimo: number;
    puntoReorden: number;
    estado: boolean;
    saldo: number;
    usuarioCreador: string;
    usuarioModificador: string;
    index: number;
    codigoDescripcion: string;
}

export class CatalogoBienFilter {
    pageNumber: number;
    pageSize: number;
    sortColumn: string;
    sortOrder: string;
    clasificadorIngresoId: number;
    codigo: string;
    descripcion: string;
    estado: boolean;
}

export class CatalogoBienSaldo {
    catalogoBienId: number;
    clasificadorIngresoId: number;
    unidadMedidaId: number;
    unidadMedida: any;
    codigo: string;
    descripcion: string;
    estado: boolean;
    saldo: number;
    index: number;
}

File: recaudacion/2.Codigo/backend/RecaudacionApiIngresoPecosa/Domain/CatalogoBien.cs
namespace RecaudacionApiIngresoPecosa.Domain
{
    public class CatalogoBien
    {
        public int CatalogoBienId { get; set; }
        public int ClasificadorIngresoId { get; set; }
        public int UnidadMedidaId { get; set; }
        public UnidadMedida UnidadMedida { get; set; }
        public string Codigo { get; set; }
        public string Descripcion { get; set; }
        public int? StockMaximo { get; set; }
        public int? StockMinimo { get; set; }
        public int? PuntoReorden { get; set; }
        public bool Estado { get; set; }
        public int? Saldo { get; set; }
        public string UsuarioCreador { get; set; }
        public string UsuarioModificador { get; set; }
    }
}